packages/e2e/test/load-test-custom/wallet-init/wallet-init.test.ts

```diff
@@ -0,0 +1,112 @@
+    env.NETWORK_INFO_PROVIDER,
+    env.NETWORK_INFO_PROVIDER_PARAMS,
+    logger
+  ),
+  rewardsProvider: await rewardsProviderFactory.create(env.REWARDS_PROVIDER, env.REWARDS_PROVIDER_PARAMS, logger),
+  stakePoolProvider: await stakePoolProviderFactory.create(
+    env.STAKE_POOL_PROVIDER,
+    env.STAKE_POOL_PROVIDER_PARAMS,
+    logger
+  ),
+  txSubmitProvider: await txSubmitProviderFactory.create(env.TX_SUBMIT_PROVIDER, env.TX_SUBMIT_PROVIDER_PARAMS, logger),
+  utxoProvider: await utxoProviderFactory.create(env.UTXO_PROVIDER, env.UTXO_PROVIDER_PARAMS, logger)
+});
+
+const getKeyAgent = async (accountIndex: number) => {
+  const createKeyAgent = await keyManagementFactory.create(
+    env.KEY_MANAGEMENT_PROVIDER,
+    { ...env.KEY_MANAGEMENT_PARAMS, accountIndex },
+    logger
+  );
+  const walletUtil = createLazyWalletUtil();
+  const keyAgent = await createKeyAgent({ inputResolver: walletUtil });
+  return { keyAgent, walletUtil };
+};
+
+const createWallet = async (accountIndex: number): Promise<SingleAddressWallet> => {
+  measurementUtil.addStartMarker(MeasureTarget.keyAgent, accountIndex);
+  const providers = await getProviders();
+  const { keyAgent, walletUtil } = await getKeyAgent(accountIndex);
+  measurementUtil.addMeasureMarker(MeasureTarget.keyAgent, accountIndex);
+
+  measurementUtil.addStartMarker(MeasureTarget.wallet, accountIndex);
+  const wallet = new SingleAddressWallet({ name: `Wallet ${accountIndex}` }, { ...providers, keyAgent, logger });
+  walletUtil.initialize(wallet);
+  return wallet;
+};
+
+const initWallet = async (idx: number) => {
+  const wallet = await createWallet(idx);
+  await waitForWalletStateSettle(wallet);
+  measurementUtil.addMeasureMarker(MeasureTarget.wallet, idx);
+  return wallet;
+};
+
+// A very simple print function. Measurement util returns an object that could be used in any way.
+const showResults = () => {
+  // REVIEW: replace with a logger method. Currently it fails at lint stage
+  console.log('Measurements:', measurementUtil.getMeasurements([MeasureTarget.wallet, MeasureTarget.keyAgent]));
+};
+
+// Starts observing measurement markers. If this method is not called, no measurements are done.
+measurementUtil.start();
+
+// Simple scheduler that distributes the requested number of calls evenly in the duration time
+getLoadTestScheduler<SingleAddressWallet>(
+  {
+    // callUnderTest must be a method returning an observable
+    callUnderTest: (id) => from(initWallet(id)),
+    callsPerDuration: walletCount,
+    duration
+  },
+  { logger }
+)
+  .pipe(toArray()) // Waiting until all wallets settle to call shutdown for all of them
+  .subscribe({
+    complete: () => {
+      logger.info('disconnecting');
+      showResults();
+      measurementUtil.stop();
+    },
+    // Shutdown all wallets. Remove `toArray` operator to shut them down immediately after being settled
+    next: (wallets) => wallets.map((wallet) => wallet.shutdown())
```

Comment:
Is it possible to wrap and reuse the observable shutdown logic across many similar e2e tests? (ex. wallet restoration) if we choose this approach (custom observable test scheduler)?

Reply:
The shutdown wallet is strictly related to the test needs, but you are right, we should keep an eye out for repeating shutdown sequences and extract them.
In this case it's specific to wallets and this test: wallet need shutdown() and this test intentionally shuts down all wallets at the end of the test (instead of right after settling)